greengrass_defender_agent/agent.py

```diff
@@ -103,16 +144,47 @@ def publish_metrics(ipc_client, config_changed, metrics_collector, sample_interv
         raise e
 
 
+def handle_expcetion_and_sleep(retry_action, exception, retry_time_sec, retry_count):
+    """
+    Handle exception and sleep for retry
+
+    :param retry_action: the name of retry action to print in the log
+    :param exception: the exception thrown from orginal action
+    :param retry_time_sec: number of seconds to sleep before the next retry
+    """
+    config.logger.exception(
+                "Exception occurred during the {} : {}".format(retry_action, exception)
+                )
+    config.logger.info(
+                "Will retry {} in {} seconds, retry count remaining: {}".format(retry_action, retry_time_sec, retry_count)
+            )
+    sleep(retry_time_sec)
+
+
 def main():
     # Get the ipc client
     ipc_client = ipc_utils.IPCUtils()
-    try:
-        ipc_client.connect()
-    except Exception as e:
-        config.logger.error(
-            "Exception occurred during the creation of an IPC client: {}".format(e)
-        )
-        exit(1)
+
+    # Connect to the GG Nucleus
+    need_retry = True
+    retry_time = config.INITIAL_RETRY_INTERVAL_SECONDS
+    env_config = set_env_variables_config()
+    connect_retry = config.DEFAULT_CONNECT_AND_PUBLISH_RETRY
```

Comment:
After discussed with Michael I decided to not make this retry count number configurable to customer. For connect to ipc client, it would have 5 inline retries then exit.